modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5694,13 +5696,18 @@ public void addOperationToGroup(String tag, String resourcePath, Operation opera
         }
         // check for operationId uniqueness
         String uniqueName = co.operationId;
-        int counter = 0;
+        int counter = seenOperationIds.getOrDefault(uniqueName, 0);
+        while(seenOperationIds.containsKey(uniqueName)) {
+            uniqueName = co.operationId + "_" + counter;
+            counter++;
+        }
```

Comment:
looks like this breaks the use case in which more than one operation having the same name (e.g. getId) and now users will get something different

Reply:
You can revert it but we will need a sample for the use case this breaks and another pr with this fix that doesn't break that sample.

Reply:
When you say users get something different, is it wrong, or just different?

Reply:
I think it's nice addition to prevent spec with duplicated operationId generating code that won't compile. Thanks again for the PR.

Let me think about how to deal with auto fix (opt in or out) and come up with a solution.

Will definitely add samples/tests for to cover this case moving forward.

Thanks again for your work!

Reply:
> When you say users get something different, is it wrong, or just different?

please refer to the discussion in https://openapi-generator.slack.com/archives/CLSB0U0R5/p1728581031429579?thread_ts=1728372817.754659&cid=CLSB0U0R5

Reply:
update: i've filed https://github.com/OpenAPITools/openapi-generator/pull/19872 but no idea off the top of my head why the C# (tag) sample project failed but I don't see duplicated operationId in the test spec (modules/openapi-generator/src/test/resources/3_0/csharp/tags.json)

will take another look tomorrow.

if you've any clue, please let me know

Reply:
Try opting the sample into the new normalizer option. Though im not thrilled about it not compiling without opting into a normalizer rule.

Reply:
Tried locally (not committed) but no luck

Reply:
Will take a look when i can. Maybe not till the weekend.

Reply:
please take your time

Reply:
I have multiple endpoints in my openapi.json which have the same operationId. This was perfectly fine in 7.8.0 and is also fine "in real life" because they are all below different path. I have one GET endpoint "/moneyflow/server/monthlysettlement/getAvailableMonth" with operationId "getAvailableMonth" and one GET endpoint "/moneyflow/server/report/getAvailableMonth" also with operationId "getAvailableMonth".

With the JavaSpring generator, I now get interfaces generated with "getAvailableMonth_0" inside the `ReportControllerApi` and one "getAvailableMonth" in `MonthlySettlementControllerApi`.

This is just an example - I have multiple endpoints with "create", "delete" and so on operationIds - they now all end up getting generated as create_0 up to create_4 - all in different Controllers.... this really looks like a mess to be honest.

How can I revert back to the old behaviour. There is no point to make this methods unique project-wide when they end up getting generated in different Controller Interfaces (not generating a client). Additionally those numbers don't seem to be predictable so it just ends up with not usable method names (because they may vary in the future I guess)

Reply:
Clearly the _0 is a breaking change. The issue for me is that I use the operationId in the name of a class which means it has to be unique. I thought using the operationId would give me a unique value, but that is not true when there are tags. I'll have to explore a different way to handle this.

Reply:
On class level you are right (have to be unique). If I understood the initial problem correctly, it was all about Windows filesystem limitations (case insensitive). So maybe the file names have to be unique case-insensitiive wise, but the code inside the files might not. I mean while Windows cant have File1.x and file1.x in parallel, it might have File1.x and file1_0.x but with classes File1 and file1 inside. But - not sure if there are languages out which dictate classname must be equals filename.

And one problem remains imho. The names become inpredictable. What guarantees that "...._0" will be "..._0" in the future and not silently will be generated as "..._2" at one point instead. Maybe just because some dev decides to reorder the openapi spec? Adding iterating numbers seems not right at all imho but maybe I'm missing something :smile:

Reply:
The _0 part of this pr reverted in a new pr. Thanks for being understanding. https://github.com/OpenAPITools/openapi-generator/pull/19913